File: stimes-erp-web/src/app/core/services/user-rights.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface UserRights {
  access: boolean;
  add: boolean;
  edit: boolean;
  delete: boolean;
  search: boolean;
  approve: boolean;
}

export const NO_RIGHTS: UserRights = { access: false, add: false, edit: false, delete: false, search: false, approve: false };

/**
 * Generic per-user, per-form rights client - matches the desktop app's
 * Stimes.Erp.Library.UserRights.AssignUserRights (Access/Add/Edit/Delete/Search/Approve),
 * reusable by any form's list/detail component.
 */
@Injectable({ providedIn: 'root' })
export class UserRightsService {
  private base = `${environment.apiBaseUrl}/userrights`;

  constructor(private http: HttpClient) {}

  getRights(formClassName: string): Observable<UserRights> {
    return this.http.get<UserRights>(this.base, { params: { formClassName } });
  }
}

File: StimesErp.Api/StimesErp.Api/Services/UserRightsService.cs
using System.Data;
using StimesErp.Api.Data;

namespace StimesErp.Api.Services
{
    /// <summary>
    /// Web equivalent of Stimes.Erp.Library.UserRights.AssignUserRights (backed by
    /// Stimes.Erp.Helper.Security.GetItemSecurity / usp_GetUserRightSecurity) - the same
    /// per-user, per-form Access/Add/Edit/Delete/Search/Approve rights the desktop app's
    /// CheckPermission() reads before enabling Save/New/Delete on a form.
    /// </summary>
    public class UserRightsService
    {
        private readonly SqlHelper _db;

        public UserRightsService(SqlHelper db)
        {
            _db = db;
        }

        public DataTable GetRights(string formClassName, int userCode, int systemCode)
        {
            var p = new[]
            {
                SqlHelper.Param("@tcModuleName", SqlDbType.VarChar, formClassName, 500),
                SqlHelper.Param("@tnUserCode", SqlDbType.Int, userCode),
                SqlHelper.Param("@tcAccess", SqlDbType.VarChar, "", 1),
                SqlHelper.Param("@tnSystemCode", SqlDbType.Int, systemCode)
            };
            return _db.GetDataTableFromProcedure("usp_GetUserRightSecurity", p);
        }
    }
}
